Tests/SwiftParserTest/ExpressionInterpretedAsVersionTupleTests.swift

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftParser
+import SwiftSyntax
+import _SwiftSyntaxTestSupport
+import XCTest
+
+final class ExpressionInterpretedAsVersionTupleTest: XCTestCase {
+  func testExpressionInterpretedAsVersionTuple() throws {
+    let expression: ExprSyntax = "2.2.2.2"
+    let versionTuple = try XCTUnwrap(expression.interpretAsVersionTuple())
+    let subtreeMatcher = SubtreeMatcher(Syntax(versionTuple), markers: [:])
```

Comment:
As of https://github.com/apple/swift-syntax/pull/2027, you don’t need the `Syntax` upcast here anymore. Same for a couple more `Syntax` upcasts in this file